src/katello/repos.py

```diff
@@ -68,22 +68,15 @@ def remove_cache():
             pass
 
     def is_valid(self):
-        cache_file = None
-        # wrapper try block w/ finally needed for python 2.4
         try:
-            try:
-                cache_file = open(ENABLED_REPOS_CACHE_FILE, 'r')
-                try:
-                    return self.data() == json.loads(cache_file.read())
-                except ValueError:
-                    return False
-            except IOError:
-                error = sys.exc_info()[1]  # backward and forward compatible way to get the exception
-                if error.errno == errno.ENOENT:
-                    return False
-        finally:
-            if cache_file is not None:
-                cache_file.close()
+            with open(ENABLED_REPOS_CACHE_FILE, 'r') as cache_file:
+                return self.data() == json.loads(cache_file.read())
+        except ValueError:
+            return False
+        except IOError:
+            error = sys.exc_info()[1]  # backward and forward compatible way to get the exception
+            if error.errno == errno.ENOENT:
+                return False
```

Comment:
```suggestion
        except IOError as exc:
            if exc.errno == errno.ENOENT:
                return False
```

Should work on 2.7+

Reply:
What I don't get is why this is special-cased at all -- if there was an IOError, the cache is not valid.

Reply:
Actually, if there was *any* error/exception, it's not valid. The only difference the code right now makes is returning `False` vs `None`, but nobody cares about that.

Reply:
I wasn't quite sure about it and needing Python 2.6 at the time. Did we completely drop EL6 support, including in downstream?

Reply:
Updated to simplify. How about this version?

Reply:
RHEL6 is dead-dead-dead soon: https://access.redhat.com/articles/4665701
And those changes won't land downstream before that anyway.

Reply:
What's the soon part? I see since July 1st it's in a "we provide you with docs so you can upgrade" phase.